File: dfs,bfs/programmers/targetNumber/bella.py
# def solution(numbers, target):
#     cnt = len(numbers)
#     answer = 0
#
#     def dfs(index, result):
#         if index == cnt:
#             if result == target:
#                 nonlocal answer
#                 answer = answer + 1
#             return
#         else:
#             dfs(index + 1, result + numbers[index])
#             dfs(index + 1, result - numbers[index])
#
#     dfs(0, 0)
#     return answer

# 고인물 코드
def solution2(numbers, target):
    if not numbers and target == 0 :
        return 1
    elif not numbers:
        return 0
    else:
        return solution2(numbers[1:], target-numbers[0]) + solution2(numbers[1:], target+numbers[0])

File: dfs,bfs/programmers/wordConversion/bella.java
import java.util.Arrays;
class Solution {
    static public boolean[] visited;
    static public int answer = 0;

    public boolean diffCheck(String begin, String target){
        int diffCnt = 0;
        for(int i = 0 ; i < begin.length(); i++)
            if(begin.charAt(i) != target.charAt(i)) diffCnt++;

        return diffCnt == 1? true:false;
    }

    public int dfs(String std, String target, String[] words, int cnt){
        if(std.equals(target)){
            answer = cnt;
            return cnt;
        }

        for(int i = 0 ; i < words.length; i++){
            if(!visited[i] && diffCheck(std,words[i])){
                visited[i] = true;
                dfs(words[i],target,words,cnt+1);
                visited[i] = false;
            }
        }
         return 0;
    }

    public int solution(String begin, String target, String[] words){
        visited = new boolean[words.length];
        if(!Arrays.asList(words).contains(target)) return answer;
        dfs(begin,target,words,0);
        return answer;
    }
}